### ISSUE TITLE
Why the keep_rate in sa is (1, )*12?

### ISSUE THREAD

**Issue by YRlin-12** [buptLinfy/ZSE-SBIR#1]
Hi, your work is really good.
But I have some questions about your code that is different from your paper.
1. Why the keep_rate in sa.py is (1, )*12? And if I change the rate < 1 for several layers, the code will not run due to the token number is decreased. 

2. Why the token selection is not in cross_attention.py?

**Comment by buptLinfy** [comment]
Thank you for following our research and asking questions! 

1. Yes, the keep_rate is 1 in `sa.py`, which is consistent with **Ours-full** in our paper. If keep_rate < 1 for several layers, the sa and ca network will still work normally. According to the settings in the paper, if keep_rate is 0.9, which means `self.keep_rate = (1, 1, 1, 0.9) + (1, 1, 0.9) + (1, 1, 0.9) + (1, 1)`, the token number is finally reduced from 196 to 144. Correspondingly, the anchor number in the rn network should also follow change, such as anchor_number=36 or even 144, as this will not significantly slow down our calculations.

2. In `ca.py`, it is the same as `sa.py`.

In addition, it should be noted that due to the lack of information, the direct training of the network with selective token converges very slowly. Therefore, we recommend to **pre-train(or warm-up)** the full model for 5-10 epochs, and then train the network under the condition of `keep_rate=0.9` and `anchor_number=144`, especially re-train the rn network. This will achieve the effect in our paper, and the time consumption of training and testing is greatly reduced.
